app/src/main/java/org/mozilla/fenix/HomeActivity.kt

```diff
@@ -242,15 +244,19 @@ open class HomeActivity : LocaleAwareAppCompatActivity(), NavHostActivity {
 
         components.core.requestInterceptor.setNavigationController(navHost.navController)
 
+        MainScope().launch(Dispatchers.Main) {
+            navGraphJob.await()
+        }
+
         StartupTimeline.onActivityCreateEndHome(this) // DO NOT MOVE ANYTHING BELOW HERE.
     }
 
-    private fun createNavGraph() {
-        val navHostFragment : NavHostFragment  = container as NavHostFragment
-        val inflater = navHostFragment.navController.navInflater
-        val graph = inflater.inflate(R.navigation.nav_graph)
-        navHostFragment.navController.graph = graph
-    }
+    private fun createNavGraph() = MainScope().async(Dispatchers.Default) {
+            val navHostFragment = container as NavHostFragment
```

Comment:
nit: indentation is wrong. should be 4 spaces in the middle and closing brace lines up with `private fun`, like a normal function